### PR TITLE
Adding the client interface and unit tests for getTenantID func

### PR DESCRIPTION
**Pull request by Vidya2606**
# Description

Added the client interface and unit tests for getTenantID func

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a  breaking change which will impact consuming tool(s).

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/providers/azure_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	"context"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/tracing"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/subscription/armsubscription"
+	mock_providers "github.com/Azure/draft/pkg/providers/mock"
+	"go.uber.org/mock/gomock"
+	"testing"
+)
+
+func TestGetTenantId(t *testing.T) {
```
Comment (bfoley13) [inline pkg/providers/azure_test.go:13]: We should have tests around the different errors/scenarios possible within the getTenantId and listTenant clients

File: pkg/providers/azure.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ type SetUpCmd struct {
 	tenantId          string
 	appObjectId       string
 	spObjectId        string
+	AzTenantClient    azTenantClient
```
Comment (bfoley13) [inline pkg/providers/azure.go:29]: if were going to have many interfaces associated with the az client, we should pull the clients into a single struct and have it be initialized.

Have an AzClient struct that will contain the sub tenantClient, appsClient, etc interfaces.

**Review by davidgamero — CHANGES_REQUESTED** [review]

File: cmd/setup-gh.go [hunk 3]
```diff
@@ -24,11 +26,24 @@ func newSetUpCmd() *cobra.Command {
...
+			if err != nil {
+				return fmt.Errorf("getting credentials: %w", err)
+			}
+
+			client, err := armsubscription.NewTenantsClient(azCred, nil)
+			if err != nil {
+				return fmt.Errorf("creating tenants client: %w", err)
+			}
+
+			sc.AzClient.AzTenantClient = client
```
[the remark this answers is not in the document]
Comment (davidgamero) [reply cmd/setup-gh.go:40]: nevermind this is a struct type not a pointer

**Review by davidgamero — APPROVED** [review]
lgtm

**Review by Tatsinnit — APPROVED** [review]
THank you and looks good to me! ☕️🙏❤️

**Review by bfoley13 — APPROVED** [review]

**Pull request merged**